modules/telemetry/include/hephaestus/telemetry/telemetry.h

```diff
@@ -0,0 +1,47 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <type_traits>
+
+#include <absl/base/thread_annotations.h>
+
+#include "hephaestus/serdes/json.h"
+#include "hephaestus/telemetry/sink.h"
+
+namespace heph::telemetry {
+
+void registerSink(ITelemetrySink* sink);
+
+void metric(const MetricEntry& log_entry);
+
+template <serdes::JSONSerializable DataT>
+void metric(const std::string& component, const std::string& tag, const DataT& data,
+            ClockT::time_point log_timestamp = ClockT::now()) {
+  const MetricEntry log_entry{
+    .component = component,
+    .tag = tag,
+    .log_timestamp = log_timestamp,
+    .json_values = serdes::serializeToJSON(data),
+  };
+
+  metric(log_entry);
+}
+
+template <typename DataT>
+  requires std::is_arithmetic_v<DataT> || std::is_same_v<DataT, std::string>
+void metric(const std::string& component, const std::string& tag, const std::string& key, const DataT& value,
```

Comment:
shall we keep the generic interface this or delete  it?

Reply:
I am strongly in favour of removing it, but I want to wait for @floriantschopp feedback if he can make a good case for it.